src/kes_dashboard_live_tiles_cli/OnlineLoader.cpp

```diff
@@ -26,19 +40,50 @@ OnlineLoader::~OnlineLoader() {
 }
 
 
-std::list<Tile> OnlineLoader::getTiles() {
-    std::list<Tile> tiles;
+list<shared_ptr<ITile>> OnlineLoader::getTiles() {
+    list<shared_ptr<ITile>> tiles;
     HTTPClient httpClient;
-    std::shared_ptr<JSON_Value> response;
+    shared_ptr<JSON_Value> response;
+    JSON_Object* responseData;
+    JSON_Array* tilesData;
+    JSON_Value* tileData;
 
     // Download feed from the KES.
-    try {
-        response = httpClient.GET(this->KES_URL);
-    } catch (HTTPRequestFailedError& error) {
-        std::cerr << error.what();
+    response = httpClient.GET(this->KES_URL);
+    responseData = json_value_get_object(response.get());
+
+    // Get the array of tile data objects from the response.
+    if (!json_object_has_value_of_type(responseData, "shares", JSONArray)) {
+        throw BrokenContractsException(
+            "Does not contain array of tiles: " +
+            string(json_serialize_to_string_pretty(response.get())));
     }
+    tilesData = json_object_get_array(responseData, "shares");
 
-    // Download each tile to a temporary directory.
+    // Create Tile objects from the feed data and download additional data.
+    for (int i = 0; i < json_array_get_count(tilesData); i++) {
+        shared_ptr<ITile> tile = make_shared<Tile>();
+        tileData = json_array_get_value(tilesData, i);
 
+        // Initialise a tile object with data from the feed.
+        if (!tile.get()->initialise(tileData)) {
+            throw BrokenContractsException(
+                "For item with data: " +
+                string(json_serialize_to_string_pretty(tileData)));
```

Comment:
I think that `json_serialise_to_string_pretty()` needs to be freed with `json_free_serialized_string()` so you are leaking here. Construct with a smart pointer:

```c++
shared_ptr<char> data_str(
    json_serialize_to_string_pretty(tileData),
    json_free_serialized_string);
```

Reply:
Great catch, thanks! 👍

Reply:
Extended the custom Exceptions to accept an optional `JSON_Value*` to serialise there rather than constructing that everywhere I throw.